format/riff/common.go

```diff
@@ -19,11 +19,11 @@ func (p path) topData() any {
 	return p[len(p)-1].data
 }
 
-func riffDecode(d *decode.D, path path, headFn func(d *decode.D, path path) (string, int64), chunkFn func(d *decode.D, id string, path path) (bool, any)) {
+func riffDecode(d *decode.D, path path, headFn func(d *decode.D, path path) (string, int64), chunkFn func(d *decode.D, id string, path path, size int64) (bool, any)) {
 	id, size := headFn(d, path)
 
 	d.FramedFn(size*8, func(d *decode.D) {
-		hasChildren, data := chunkFn(d, id, path)
+		hasChildren, data := chunkFn(d, id, path, size)
```

Comment:
This uses `FramedFn` so i think `chunkFn` will be decoding in a limited/framed bit range so i think you can use `BitsLeft`/`Len` etc to get how much is left

Reply:
I may have been doing it wrong, but that's what I tried initially, hope to get to revisit soon!

Reply:
Okok yeah in this case i think relying on framing should work, let's try that first at least